src/Components/AppShell/AppShell.js

```diff
@@ -95,14 +73,13 @@ class AppShell extends Component {
     const segment = nextProps.location.pathname.split('/')[1] || 'dashboard'
 
     // If we haven't initialized GA, do it.
-    if (!this.user) {
-      this._setupGoogleAnalytics()
+    if (!GA.isSetup) {
```

Comment:
This seems like somewhat of a code smell (cohesion/coupling) that reflects a misunderstanding or shortcoming in the way the application is being initialized. Why should the application container in the renderer process care about kicking off the google analytics initialization?

Reply:
Also is the `isSetup` property global state (it looks like it is). Ideally the GA service would be a class and this sort of state would be encapsulated in an instance of that class.

Reply:
RE your first comment: I was merely converting current architecture to enabling the feature of more generalized analytics reporting. It's a lot of work to question every little potential design choice that you (or previous author) did every time we want to implement something small. Sure this is how chaos can ensue in the future, but if we ask this question, we may as well ask why the whole application is architect the way it is.

RE your second comment: `GA` is a single object that is declared once per process. This was how  the internet suggested `singleton classes` be implemented in Javascript. So yes `isSetup` is global since `GA` is global. We can talk offline if you want about why a `singleton class` would be best for this service.